### DISCUSSION TITLE
New Models

### DISCUSSION THREAD

**Discussion by incebellipipo** [Liquid-ai/Plankton#28] [General]
Hi everyone,

I recently needed to use a UUV simulation. It seems to me everything went bananas because of the migration from ROS1 to ROS2. I need to have tools on Ubuntu 20.04 but I want to use UUV Simulator and I wanted to use ROS1 since most of the packages working properly but there is no release for noetic... so and so forth. So I decided to port ECA_A9 to Plankton, and I did. 

I want to share this port but I wonder If you accept pull requests for new vehicles? If so, are there any guides to contribute?

**Comment by incebellipipo** [comment]
I published it under a public repo. 
https://github.com/GSO-soslab/eca_a9_plankton

**Reply by jpliquid** [reply]
Hi @incebellipipo,

Sorry for the delay, the github discussion section is a bit clunky, so I hadn't see your messages before. I'm very glad that you have ported ECA_A9 to Plankton. I think that a way to integrate it in Plankton would have been to create a `vehicles` folder, with a package `eca_a9` inside, but I'm perfectly fine with the repo that you created.

I will add a section in our readme to mention external contributions like yours. 
Thank you

**Reply by incebellipipo** [reply]
So true, discussions are a little bit unusual. Do you want us to use issues instead @jpliquid?

We could have separated per-vehicle repositories like the user group @[uuvsimulator](https://github.com/uuvsimulator) did. This consistency in the design may be more welcoming for the new users.

 i.e. scripts that use UUVSimulator packages may only change the user section in the URL to migrate.
